src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -209,7 +209,27 @@ private static void BindNonScalar(this IConfiguration configuration, object inst
         {
             if (instance != null)
             {
-                foreach (PropertyInfo property in GetAllProperties(instance.GetType()))
+                List<PropertyInfo> modelProperties = GetAllProperties(instance.GetType()).ToList();
+
+                if (options.ErrorOnUnknownConfiguration)
+                {
+                    HashSet<string> propertyNames = new(modelProperties.Select(mp => mp.Name),
+                        StringComparer.OrdinalIgnoreCase);
+
+                    List<string> missingPropertyNames = configuration.GetChildren()
+                        .Where(cs => !propertyNames.Contains(cs.Key))
+                        .Select(mp => $"'{mp.Key}'")
+                        .ToList();
+
+                    if (missingPropertyNames.Any())
```

Comment:
You are creating a `List` for these missing properties, you could use `Length` instead and avoid Linq.

Reply:
I've used the `Count` property.  After about 15 years of .NET, I too still cannot remember whether it's `.Length` or `.Count` :)  Luckily, ReSharper knows!